_posts/2021-06-01-migration-v1-v2.md

```diff
@@ -0,0 +1,130 @@
+
+## Migration
+
+The migration from Apicurio Registry 1.3.x to 2.x requires you to move the data from your old registry to another along with a review of your applications that interact with the registry in order to update the configuration of your applications to accomadate to the new requirements.
+
+### Data Migration
+
+The process of data migration to Apicurio Registry 2.x will require you to export all the data from your 1.3.x deployment and import it to the new 2.x based deployment.
+
+Data migration is a critical step in the migration to Apicurio Registry 2.x if you are using the registry as a schema registry for Kafka applications, because every Kafka message carries the global identifier for the schema stored in Apicurio Registry. It is critical that this identifier is kept between registry upgrades via data migration.
+
+For that purpose Apicurio Registry 2.x provides a new import/export API to bulk import or export all the data from your registry deployment. This API guarantees that all the identifiers are kept when importing the data from your old registry. The export API works by downloading a custom zip file containing all the information of your artifacts. The import API accepts that zip file, loading all of the artifacts into the registry in a single batch.
+
+Apicurio Registry 1.3.x does not provide an import/export API but with the 2.x release we provided an export tool compatible with Apicurio Registry 1.3.x that allows you to export a compatible zip file which then can be imported to your 2.x registry.  This special export tool uses common existing APIs to export all of the content in the registry.  It is less performant than the built-in export API we have introduced in 2.x, and so should only be used when exporting from a 1.3.x registry.
+
+This export tool can be found [here](https://github.com/Apicurio/apicurio-registry/tree/2.0.x/utils/exportV1). It is a java application meant to be invoked from the command line.
+
+The migration steps for moving all your data from one version to another:
+
+**Pre-requisites**: having running Apicurio Registry instances for both the version 1.3.x server (exporting from) and the new 2.x server (importing to).
+
+1. Export all the data from Apicurio Registry 1.3.x using the `exportV1` tool. This will generate a `registry-export.zip` file in your current directory.
+
+```
+java -jar apicurio-registry-utils-exportV1-2.0.0-SNAPSHOT-runner.jar http://old-registry.my-company.com/api
+```
+
+2 - Import the zip file to Apicurio Registry 2.x using the import API. Extended documentation on this API can be found [here](https://www.apicur.io/registry/docs/apicurio-registry/2.0.0.Final/getting-started/assembly-managing-registry-artifacts-api.html#exporting-importing-using-rest-api)
```

Comment:
Suggest "You can find more details in the [Apicurio Registry user documentation] (...)"